java/lite/proguard.pgcfg

```diff
@@ -2,7 +2,3 @@
 # Skip runtime check for isOnAndroidDevice().
 # One line to make it easy to remove with sed.
 -assumevalues class com.google.protobuf.Android { static boolean ASSUME_ANDROID return true; }
-
--keepclassmembers class * extends com.google.protobuf.GeneratedMessageLite {
```

Comment:
does this also need to be removed from here, other things may be depending on this? Also, the -assumevalues bit, should that exist in the new file?

Reply:
I believe the assumevalues is something related to this build and testing. https://github.com/protocolbuffers/protobuf/blob/main/java/core/src/main/java/com/google/protobuf/Android.java#L39

The point of this change is to put the config we want all apps to get into the inherited location. See https://twitter.com/jakewharton/status/1004401938467876865

I hope nothing should rely on the keepclassmembers from here, because hopefully everything is picking up the new proguard file in the right location.  But I'd like to flush out errors to prove this is true.

Reply:
cc @agrieve Not sure if you can provide insight into where the ASSUME_ANDROID proguard rules should live? Are they required to be shipped with the published libraries?

Reply:
I've added that file. Intellij gives a lot of errors for it.

I've removed the original file, because I think we want to see that it's building and passing tests with the new file, not saved by the old file.

But happy to revisit.